omegaconf/basecontainer.py

```diff
@@ -784,7 +784,7 @@ def _create_structured_with_missing_fields(
 def _is_missing_value(value: Any) -> bool:
     if isinstance(value, Container):
         value = value._value()
-    ret = value == "???"
+    ret = value == "???" if isinstance(value, str) else False
```

Comment:
So the idea is to sidestep the custom `DataFrame.__eq__` method by first deciding if `value` is a `str`, then only calling `value.__eq__("???")` if we are sure that `value` is a `str`.

Reply:
yep, exactly.